extensions/wikia/UserLogin/FacebookSignupController.class.php

```diff
@@ -78,7 +78,10 @@ public function index() {
 				// Retrieve user email from Facebook if missing
 				$email = $user->getEmail();
 				if ( empty( $email ) ) {
-					$this->saveEmailAsynchronously( $user->getId() );
+					$this->saveEmailAsynchronously( $user->getId(), true );
+				} else {
+					// Send welcome email
+					F::app()->sendRequest( 'Email\Controller\Welcome', 'handle', [] );
```

Comment:
You know I'm a fan of class constants for this kind of stuff :) If you wanted to pull this out into one, that would be super cool.